src/nvim/testdir/test_mksession.vim

```diff
@@ -399,6 +413,54 @@ func Test_mkview_no_file_name()
   %bwipe
 endfunc
 
+func Test_mkview_loadview_jumplist()
+  set viewdir=Xviewdir
+  au BufWinLeave * silent mkview
+
+  edit Xfile1
+  call setline(1, ['a', 'bbbbbbb', 'c'])
+  normal j3l
+  call assert_equal([2, 4], getcurpos()[1:2])
+  write
+
+  edit Xfile2
+  call setline(1, ['d', 'eeeeeee', 'f'])
+  normal j5l
+  call assert_equal([2, 6], getcurpos()[1:2])
+  write
+
+  edit Xfile3
+  call setline(1, ['g', 'h', 'iiiii'])
+  normal jj3l
+  call assert_equal([3, 4], getcurpos()[1:2])
+  write
+
+  au BufWinEnter * silent loadview
```

Comment:
This is what you moved from the line after `au BufWinLeave * silent mkview`. Leave a comment here to explain why this is needed and restore this line in its original spot but comment it out for reference.

Is this still needed?

Reply:
Still seems so.

Because `:loadview` is triggered on `BufWinEnter`, it'll run on the initial `:edit`s before the views are saved with `:mkview`, which triggers on `BufWinLeave`.

In Vim, `:loadview` silently fails if the view isn't found (doesn't use the return code from `do_source()`).
Nvim checks `do_source()`'s return code and raises `e_notopen` (E484) instead.

The lines above are only to create the views, `:loadview` is only needed after the views exist.
I'll include the original line as a comment and add the reasoning. :+1:

Also, on another note, #13490 still happens if the `:loadview` errors inside the `:au`.
It'll still cause the test to fail (from the leak + ASAN) if this were to happen, but that probably isn't ideal. 😄